parser.py

```diff
@@ -23,13 +25,13 @@ def parse_url(self, url):
         if not match:
             return False
 
-        protocol = match.group('protocol')
-        itemtype = match.group('type')
-        host = match.group('host')
-        port = match.group('port')
-        resource = match.group('resource')
-
-        if not host:
+        try:
+            protocol = match.group('protocol')
+            itemtype = match.group('type')
+            host = match.group('host')
+            port = match.group('port')
+            resource = match.group('resource')
+        except:
```

Comment:
I couldn't remember what the error was here so I did this for expediency. It is a specific error that has cropped up a few times when the url does not match the regex... I'll have to intentionally trigger it later and add it as an explicit except here.